src/verify.c

```diff
@@ -1063,6 +1115,23 @@ enum swupd_code execute_verify(void)
 	}
 	timelist_timer_stop(globals.global_times);
 
+	/* get the list of files to verify */
+	files_to_verify = official_manifest->files;
+	if (cmdline_option_file) {
+		/* the user specified a file or path to verify, use that instead */
+		files_to_verify = keep_matching_path(official_manifest->files);
+		files_to_verify = list_sort(files_to_verify, file_sort_filename);
```

Comment:
🤔 I don't remember why I sorted it from the top of mi mind, I opened this PR some days ago. From glancing at this it would seem like it is not necessary to sort the list, but I'll take a deeper look and will remove that if not necessary.

Reply:
after looking deeper into this, looks like you are right and this list is already sorted, but it is getting reversed in `keep_matching_path()`. Having the list sorted is important so the repair/diagnose works correctly, but instead of sorting again I will just modify `keep_matching_path()` so it preserves the order.